### ISSUE TITLE
Support Android Gradle Plugin 7.x

### ISSUE THREAD

**Issue by yujincheng08** [google/protobuf-gradle-plugin#540]
AGP 7.x has updated its API drastically. For example, `com.android.build.gradle.api.AndroidSourceSet` has renamed to `com.android.build.api.dsl.AndroidSourceSet` and thus #433 no longer works.

**Comment by yujincheng08** [comment]
@voidzcy could you please take a look at it? My current workaround is adding the following:
```
fun com.android.build.api.dsl.AndroidSourceSet.proto(action: SourceDirectorySet.() -> Unit) {
    (this as? ExtensionAware)
        ?.extensions
        ?.getByName("proto")
        ?.let { it as? SourceDirectorySet }
        ?.apply(action)
}
```

**Comment by Cloudant-android** [comment]
I also encountered this problem

**Comment by voidzcy** [comment]
I think we could drop support for AGP version < 4. Some non-trivial cleanups may be needed.

**Comment by yujincheng08** [comment]
Any progress on this?

**Comment by yujincheng08** [comment]
seems it has been fixed by 6a71480c38d1a6c050de0a1ec7437459bdb32593
